### DISCUSSION TITLE
Equivalent of CaImAn-MATLAB field "Df"?

### DISCUSSION THREAD

**Discussion by mitlabence** [flatironinstitute/CaImAn#1291] [Q&A]
Hey everyone,
I am in the process of porting ancient Matlab code that uses CaImAn-MATLAB to python. In this code, I came across the field/attribute [`Df`](https://github.com/flatironinstitute/CaImAn-MATLAB/blob/master/%40CNMF/CNMF.m#L27) being used to normalize `C` (temporal components) for each detected cell (I believe this is equivalent to the `C_df` field in the old CaImAn). I thought [`F_dff`](https://github.com/flatironinstitute/CaImAn/blob/main/caiman/source_extraction/cnmf/estimates.py#L56) is the modern alternative to this normalization, but it seems like the latter uses the deconvolved signal, while the former `C/Df` uses the temporal component. Did I miss something? Is there an alternative to `Df` that I can use to normalize the temporal components?
Thank you for your help!

**Comment by EricThomson** [comment]
Egads I would recommend you just start using Caiman, which has been steadily being updated with debugging etc over the years and offers support etc, rather than rewrite your old Matlab code in Python (Caiman is basically that Caiman-Matlab rewritten in Python, incidentally 😄 ). 

But _if you insist_:  https://github.com/flatironinstitute/CaImAn/blob/c873b036135a4c2f441c78449d42eb887d356b7e/caiman/source_extraction/cnmf/utilities.py#L442

You will find the `Df` you are after in there, if you must. 😨 This is the background normalizing term in the denominator Caiman uses for dff calculation.  Good luck.
